src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,320 @@
+
+        assertNotEquals(user1, user2);
+    }
+
+    @Test
+    public void test_hashCode_equalUsers_equalHashCodes() {
+        User user1 = new User();
+        user1.setLogin("user1");
+        user1.setPassword("password");
+        user1.setAge(MIN_AGE);
+
+        User user2 = new User();
+        user2.setLogin("user1");
+        user2.setPassword("password");
+        user2.setAge(MIN_AGE);
+
+        assertEquals(user1.hashCode(), user2.hashCode());
+    }
+
+    @Test
+    public void test_hashCode_differentUsers_differentHashCodes() {
+        User user1 = new User();
+        user1.setLogin("user1");
+        user1.setPassword("password");
+        user1.setAge(MIN_AGE);
+
+        User user2 = new User();
+        user2.setLogin("user2");
+        user2.setPassword("password");
+        user2.setAge(MIN_AGE);
+
+        assertNotEquals(user1.hashCode(), user2.hashCode());
+    }
+
+    @Test
+    public void test_gettersAndSetters_workOk() {
+        User user = new User();
+        user.setLogin("user1");
+        user.setPassword("password");
+        user.setAge(MIN_AGE);
+        user.setId(1L);
+
+        assertEquals("user1", user.getLogin());
+        assertEquals("password", user.getPassword());
+        assertEquals(MIN_AGE, user.getAge());
+        assertEquals(FIRST_USER_ID, user.getId());
+
+        user.setLogin("newUser");
+        user.setPassword("newPassword");
+        user.setAge(MAX_AGE);
+        user.setId(2L);
+
+        assertEquals("newUser", user.getLogin());
+        assertEquals("newPassword", user.getPassword());
+        assertEquals(MAX_AGE, user.getAge());
+        assertEquals(SECOND_USER_ID, user.getId());
+    }
+
+    /* Tests in class RegistrationServiceImpl */
+
+    @Test
+    public void register_existingUser_throwsException() {
+        User user1 = new User();
+        user1.setLogin("Recruit");
+        user1.setPassword("password");
+        user1.setAge(MIN_AGE);
+        registrationService.register(user1);
+
+        User user2 = new User();
+        user2.setLogin("Recruit");
+        user2.setPassword("KingBoss");
+        user2.setAge(MIN_AGE);
+
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user2));
+    }
+
+    @Test
+    public void register_loginLength_lessThan6_throwsException() {
+        User user = new User();
+        user.setLogin("PON");
+        user.setPassword("MyCode");
+        user.setAge(MIN_AGE);
+
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_loginLength_6_isOK() {
+        User user = new User();
+        user.setLogin("Peters");
+        user.setPassword("passCodes");
+        user.setAge(MIN_AGE);
+
+        assertDoesNotThrow(() -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_loginLength_moreThan6_isOK() {
+        User user = new User();
+        user.setLogin("NancyHale");
```

Comment:
test length of exactly 6
same for passwords - test 5/6, as you do for age - 17/18